Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseEvaluation/UnreachableCaseInspector.cs

```diff
@@ -98,23 +98,23 @@ internal class UnreachableCaseInspector : IUnreachableCaseInspector
 
         private UnreachableCaseInspection.CaseInspectionResultType? InvalidRangeExpressionsType(ICollection<IRangeClauseExpression> rangeClauseExpressions)
         {
-            if (rangeClauseExpressions.Any(expr => expr.IsMismatch))
+            if (rangeClauseExpressions.Any(expr => expr?.IsMismatch == true))
```

Comment:
All the conditions have been changed, making the original comment outdated.
However, it still remains that the numerous `expr?` should be converted to one guard clause at the start.

Reply:
I guess I am too tired right now. I did not see that they are all in `Any` methods. 
That being said, I actually find `expr?.IsMismatch == true` harder to parse than `expr != null && expr.IsMismatch`.

Actually, you could just define `var rangeClauses = rangeClauseExpressions.Where(expr => expr != null)`, which is safe to iterate over multiple times since you know that it is based on a collection.

Reply:
To be fair I also missed the `Any` methods in my original response to Ben ;-)

Reply:
I went with the last option, pulling the usable clauses into their own query, materialized into a `List` to avoid iterating all expressions when it's not needed.